community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/ConcurrentMapState.java

```diff
@@ -73,19 +70,19 @@ public String toString()
     @Override
     public EntryUpdater<Key> updater( long version, Lock lock )
     {
-        if ( versionSequence.seen( version, META ) )
+        if ( version <= previousVersion )
         {
-            throw new IllegalStateException( "Cannot apply update with given version " + version +
-                                             " when base version is " + previousVersion );
+            return EntryUpdater.noUpdates();
         }
-        versionSequence.offer( version, META );
         update( highestAppliedVersion, version );
+        hasTrackedChanges.set( true );
```

Comment:
`hasTrackedChanges` is used to determine  if we have any changes that satisfy `version > threshold` - meaning changes that came in a post state during rotation. And as soon as we have at least one - our state hasChanges. If we do not have any - all changes where from previous state